### ISSUE TITLE
Map writes are not thread safe 

### ISSUE THREAD

**Issue by novalogic7** [chargebee/chargebee-go#16]
we are using chargebee-go@v2.1.1 and it crashes our application.

```
fatal error: concurrent map writes

goroutine 157538 [running]:
runtime.throw(0x2101197, 0x15)
        /usr/local/go/src/runtime/panic.go:774 +0x72 fp=0xc000a9e3c8 sp=0xc000a9e398 pc=0x430d12
runtime.mapassign_faststr(0x1d88060, 0xc001370510, 0xc0015104f0, 0x7, 0x414ca2)
        /usr/local/go/src/runtime/map_faststr.go:291 +0x3fe fp=0xc000a9e430 sp=0xc000a9e3c8 pc=0x41503e
github.com/chargebee/chargebee-go.parseMap(0xc001c47d10, 0x0, 0x0, 0x0, 0x0, 0xc001c47d70)
        /go/pkg/mod/github.com/chargebee/chargebee-go@v2.1.1+incompatible/util.go:74 +0x196 fp=0xc000a9e5f0 sp=0xc000a9e430 pc=0xdf4126
github.com/chargebee/chargebee-go.SerializeParams(0x1c504a0, 0xc000ca2b60, 0xc000a9e750)
        /go/pkg/mod/github.com/chargebee/chargebee-go@v2.1.1+incompatible/util.go:40 +0x1f5 fp=0xc000a9e708 sp=0xc000a9e5f0 pc=0xdf3d05
github.com/chargebee/chargebee-go.Send(...)
        /go/pkg/mod/github.com/chargebee/chargebee-go@v2.1.1+incompatible/http_util.go:30
github.com/chargebee/chargebee-go/actions/subscription.Update(0xc0008e0c80, 0x11, 0xc000ca2b60, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /go/pkg/mod/github.com/chargebee/chargebee-go@v2.1.1+incompatible/actions/subscription/subscription.go:40 +0xec fp=0xc000a9e7a0 sp=0xc000a9e708 pc=0xe3735c
```

**Comment by cb-prajaktachavan** [comment]
Hi, Could you please share exact code snippet.

**Comment by cb-khushbubibay** [comment]
Hi @novalogic7 
This issue has been addressed in v2.13.0.
Request you to please use v2.13.0 or above version of chargebee-go SDK.
